src/collection-view.js

```diff
@@ -46,33 +50,42 @@ Marionette.CollectionView = Marionette.View.extend({
   },
 
   endBuffering: function() {
+    // Only trigger attach if already shown and attached, otherwise Region#show() handles this.
+    var canTriggerAttach = this._isShown && Marionette.isNodeAttached(this.el);
+    var nestedViews;
+
     this.isBuffering = false;
-    this._triggerBeforeShowBufferedChildren();
 
-    this.attachBuffer(this);
+    if (this._isShown) {
```

Comment:
During the show cycle of a CollectionView being shown in a Region, the following occurs in sequence:
1. collectionView.render()
  * child `before:render`
  * child `render`
1. trigger collectionView `before:show`
  * childView `before:show`
1. trigger on collectionView and descendants `before:attach` if Region is attached to DOM
  * childView `before:attach`
1. append collectionView html to region
1. trigger on collectionView and descendants `attach` if Region is attached to DOM
  * childView `attach`
1. trigger collectionView trigger `show`
  * childView `show`
1. _isShown is set to `true` [here](https://github.com/marionettejs/backbone.marionette/blob/08bfcbea6000901d8a268ad572883090c0c85868/src/dom-refresh.js#L13)

It is expected that a CollectionView that is already rendered will fire `before:show`/`show` and `before:attach`/`attach` on its children as they are added to a CollectionView that is already showing, or the same CollectionView is re-rendered manually.  It is also expected that a Region will instigate these events when showing a CollectionView, and in the attach case the Region gains a hard reference to every descendant view.  Without the _isShown check, these two cases have an overlap: during a Region's show cycle, both the CollectionView and Region would be responsible for calling attach/show events.